### DISCUSSION TITLE
Best practices for a php docker image

### DISCUSSION THREAD

**Discussion by cdudevops** [BretFisher/ama#228] [Q&A]
Hello Bret,

I follow you since a long time and I like very much to watch your courses and presentations.

In my company we have lots of web apps (apache,php,mysql)  installed on physical hosts and I want to migrate them to docker, to move in this direction.

Can you provide me some best practices/guidelines for docker apache/nginx + php image ? Should I keep them separately or in the same container ? Which php image will be better: alpine, official php image, ubuntu+php ? 

I saw your presentation about Nodejs Rocks in Docker, maybe we can see something for PHP :).

Thanks!

**Comment by BretFisher** [comment]
A bit outdated way to combine into a single container image: https://github.com/BretFisher/php-docker-good-defaults

That's what you have to do in Docker/Swarm because it doesn't have a way to link multiple containers together for non TCP/IP traffic (like sockets).

Kubernetes *does* support this with Pods, which can be 2 containers with the same process list, volumes (file sockets), and network namespace (localhost). I don't have an example of that but it would be using something like php-fpm + nginx official images out of the box (which is my preferred way).
